### DISCUSSION TITLE
Is it possible to proxy @planetscale/database initialised with client url?

### DISCUSSION THREAD

**Discussion by Izhaki** [planetscale/discussion#395] [Q&A] [answered]
We run the proxy like so:

```shell
pscale connect my-db main --port 3309
```

We create the client like so:

```js
const client = new Client({
  url: 'mysql://root@127.0.0.1:3309/my-db',
});
```

We get:
```shell
 [TypeError: fetch failed] {
  cause:  [Error: connect ECONNREFUSED 127.0.0.1:443] {
  errno: -61,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 443
}
```

**Comment by jgreet** [comment, marked as answer]
Hi @Izhaki,

`pscale connect` creates a proxy for tunneling traditional MySQL connections to PlanetScale while the databasejs driver is intended to [connect via secure HTTP in constrained environments](https://planetscale.com/blog/introducing-the-planetscale-serverless-driver-for-javascript) where traditional database connections are not possible.  The two are not compatible.

If you want to go with the PlanetScale serverless driver, [generate a new password](https://planetscale.com/docs/concepts/connection-strings) and use those credentials to [configure the client](https://github.com/planetscale/database-js#database-url). 

I also wanted to note that we no longer recommend using `pscale connect` proxies in production. At one point it was the only way to connect to a PlanetScale database but that hasn't been the case since we added [traditional connection strings](https://planetscale.com/blog/connect-any-mysql-client-to-planetscale-using-connection-strings).  At this point we would only recommend using `pscale connect` in a local dev environment when you don't want to keep track of credentials for multiple branches, or when a database client lacks support for secure connections.

**Reply by Izhaki** [reply]
Thanks for the reply!

When I used the Prisma Client, I used `prisma connect` on local to tunnel to the development DB.

Been using connection string for anything but local.

Thing is, you have to put this connection string on your machine (in my case a git-ignored `.env` file). Still slightly safer to tunnel through the proxy.

All good! Just wanted to confirm.

Perhaps it is worth leaving a note on the documentation?

**Reply by jgreet** [reply]
> you have to put this connection string on your machine.  Still slightly safer to tunnel through the proxy.

Thanks for mentioning that, I see why it would be useful. We may want to look at adding support for proxying serverless driver connections  in the future.

**Reply by itsjxck** [reply]
This feels like a bad DX decision. If someone wants to use `@planetscale/database` they lose a lot of the useful management provided by the Planetscale CLI. The whole branching workflow falls back to tedious manual management and sharing of credentials and updating local env files with connection strings/credentials, lest your account is flooded with credentials for any and all branches. You've already solved this problem in a fantastic way with CLI tooling, especially for teams, which entirely removes the need to manage connection strings during dev, but to make your own tooling incompatible with another fantastic tool you've developed feels like an incredible oversight. 

I understand your point around "the databasejs driver is intended to connect via secure HTTP in constrained environments", but we should still be able to develop with it in un-constrained environments leveraging the other tooling available for Planetscale.

**Reply by mscoutermarsh** [reply]
@itsjxck We have an issue here tracking the problem: https://github.com/planetscale/database-js/issues/110
